### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by R0SSI89** [overviewer/Minecraft-Overviewer#1483]
After a year, I'm trying to setup my Minecraft server again and was trying to use my old configuration for generating maps. But it generated a "Segmentation fault" error right after "Generating textures...". So I tried it with a default configuration (only setting a output dir, texture path and a basic world setting), but also with this an "Segmentation fault".

Overviewer is installed via the Ubuntu Repo; apt install minecraft-overviewer

My old configuration file (/opt/mscs/worlds/survival/overviewer-settings.py):
```
processes = 4

outputdir = "/var/www/map"
texturepath = "/opt/mscs/clients/client_1.12.2.jar"

worlds["survival"] = "/opt/mscs/backups/survival/survival"

renders["survivalday"] = {
    "world": "survival",
    "title": "Survival Daytime",
    "rendermode": smooth_lighting,
    "dimension": "overworld",
}

renders["survivalnight"] = {
    "world": "survival",
    "title": "Survival Nighttime",
    "rendermode": smooth_night,
    "dimension": "overworld",
}

renders["survivalcave"] = {
    "world": "survival",
    "title": "Survival Cave",
    "rendermode": [Base(), EdgeLines(opacity=0.2), Cave(only_lit=True), SmoothLighting(strength=0.5)],
    "dimension": "overworld",
}
```

Basic configuration file:
```
processes = 4

outputdir = "/var/www/map"
texturepath = "/opt/mscs/clients/client_1.12.2.jar"

worlds["survival"] = "/opt/mscs/backups/survival/survival"

renders["survivalday"] = {
    "world": "survival",
    "title": "Survival Daytime",
    "rendermode": 'smooth_lighting'
}
```

gdb output:
```
root@mcserver:/opt/mscs# gdb python
GNU gdb (Ubuntu 8.1-0ubuntu3) 8.1.0.20180409-git
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from python...(no debugging symbols found)...done.
(gdb) run /usr/bin/overviewer.py --config=/opt/mscs/worlds/survival/overviewer-settings.py -v
Starting program: /usr/bin/python /usr/bin/overviewer.py --config=/opt/mscs/worlds/survival/overviewer-settings.py -v
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
2018-09-23 16:45:17 W You are running Overviewer as root. It is recommended that you never do this, as it is dangerous for your system. If you are running into permission errors, fix your file/directory permissions instead. Overviewer does not need access to critical system resources and therefore does not require root access.
overviewer.py:342  24090 2018-09-23 16:45:17 INFO     Welcome to Minecraft Overviewer!
overviewer.py:343  24090 2018-09-23 16:45:17 DEBUG    Current log level: 10
files.py:43        24090 2018-09-23 16:45:17 DEBUG    Detected that chmods work in '/var/www/map'
files.py:59        24090 2018-09-23 16:45:17 DEBUG    Detected that overwriting renames work in '/var/www/map'
assetmanager.py:61 24090 2018-09-23 16:45:17 DEBUG    Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/overviewer_core/assetmanager.py", line 53, in __init__
    with open(config_loc) as c:
IOError: [Errno 2] No such file or directory: '/var/www/map/overviewerConfig.js'

assetmanager.py:72 24090 2018-09-23 16:45:17 DEBUG    Preferred enoding set to: UTF-8
overviewer.py:462  24090 2018-09-23 16:45:17 DEBUG    Found the following render thing: OrderedDict([('manualpois', []), ('minzoom', 0), ('rerenderprob', 0.0), ('showlocationmarker', True), ('poititle', 'Markers'), ('title', 'Survival Daytime'), ('markers', []), ('bgcolor', (26, 26, 26, 0)), ('showspawn', True), ('northdirection', 0), ('defaultzoom', 1), ('imgformat', 'png'), ('base', ''), ('world', '/opt/mscs/backups/survival/survival'), ('optimizeimg', []), ('imgquality', 95), ('overlay', []), ('rendermode', [<overviewer_core.rendermodes.Base object at 0x7ffff21b20d0>, <overviewer_core.rendermodes.EdgeLines object at 0x7ffff21b2110>, <overviewer_core.rendermodes.SmoothLighting object at 0x7ffff21b2150>]), ('dimension', ('overworld', None)), ('texturepath', '/opt/mscs/clients/client_1.12.2.jar'), ('worldname_orig', 'survival')])
world.py:275       24090 2018-09-23 16:45:17 DEBUG    regiondir is '/opt/mscs/backups/survival/survival/DIM-1/region'
world.py:276       24090 2018-09-23 16:45:17 DEBUG    rel is 'DIM-1/region'
world.py:288       24090 2018-09-23 16:45:17 DEBUG    Scanning regions.  Type is 'DIM-1'
world.py:549       24090 2018-09-23 16:45:17 DEBUG    regiondir is /opt/mscs/backups/survival/survival/DIM-1/region, has type 'DIM-1'
world.py:301       24090 2018-09-23 16:45:17 DEBUG    Done scanning regions
world.py:275       24090 2018-09-23 16:45:17 DEBUG    regiondir is '/opt/mscs/backups/survival/survival/region'
world.py:276       24090 2018-09-23 16:45:17 DEBUG    rel is 'region'
world.py:288       24090 2018-09-23 16:45:17 DEBUG    Scanning regions.  Type is None
world.py:549       24090 2018-09-23 16:45:17 DEBUG    regiondir is /opt/mscs/backups/survival/survival/region, has type None
world.py:301       24090 2018-09-23 16:45:17 DEBUG    Done scanning regions
overviewer.py:485  24090 2018-09-23 16:45:17 INFO     Generating textures...

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff668d900 in alpha_over_full () from /usr/lib/python2.7/dist-packages/overviewer_core/c_overviewer.so
(gdb) bt
#0  0x00007ffff668d900 in alpha_over_full () from /usr/lib/python2.7/dist-packages/overviewer_core/c_overviewer.so
#1  0x00007ffff668da88 in alpha_over_wrap () from /usr/lib/python2.7/dist-packages/overviewer_core/c_overviewer.so
#2  0x000055555564901a in PyEval_EvalFrameEx ()
#3  0x000055555564dd72 in PyEval_EvalFrameEx ()
#4  0x000055555564dd72 in PyEval_EvalFrameEx ()
#5  0x00005555556468ca in PyEval_EvalCodeEx ()
#6  0x000055555564e7d3 in PyEval_EvalFrameEx ()
#7  0x00005555556468ca in PyEval_EvalCodeEx ()
#8  0x00005555556461e9 in PyEval_EvalCode ()
#9  0x0000555555676bdf in ?? ()
#10 0x0000555555671952 in PyRun_FileExFlags ()
#11 0x0000555555670dcd in PyRun_SimpleFileExFlags ()
#12 0x000055555562058b in Py_Main ()
#13 0x00007ffff7a05b97 in __libc_start_main (main=0x55555561fef0 <main>, argc=4, argv=0x7fffffffe5b8, init=<optimized out>, fini=<optimized out>,
    rtld_fini=<optimized out>, stack_end=0x7fffffffe5a8) at ../csu/libc-start.c:310
#14 0x000055555561fe0a in _start ()
(gdb)
```

Those files are created in destination folder:
```
root@mcserver:/opt/mscs/worlds/survival# ls -alh /var/www/map
total 24K
drwxr-xr-x 2 minecraft minecraft 4.0K Sep 23 17:02 .
drwxr-xr-x 4 root      root      4.0K Sep 21 00:44 ..
-rw-r--r-- 1 minecraft minecraft  191 Sep 23 16:39 baseMarkers.js
-rw-r--r-- 1 minecraft minecraft   18 Sep 23 16:39 markersDB.js
-rw-r--r-- 1 minecraft minecraft   16 Sep 23 16:39 markers.js
-rw-r--r-- 1 minecraft minecraft   40 Sep 23 16:39 uuidcache.dat
```

Server info:
```
root@mcserver:/opt/mscs# uname -a
Linux mcserver 4.15.0-34-generic #37-Ubuntu SMP Mon Aug 27 15:21:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

root@mcserver:/opt/mscs# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic

root@mcserver:/opt/mscs/worlds/survival# overviewer.py -V
Minecraft Overviewer 0.12.272 (3cceaf5)
built on Mon Sep 17 21:49:02 2018
```

**Comment by twoprops** [comment]
Have you seen this? https://github.com/overviewer/Minecraft-Overviewer/issues/1455

Briefly, there is an incompatibility between the version of Pillow that ships with Ubuntu 18.04 and Overviewer. The solution is to downgrade Pillow:

`sudo pip install 'Pillow<4.3.0'`

I'm not enough of a Python player to know what consequences this might have for other Python/Pillow users on your system.

--2p

**Comment by R0SSI89** [comment]
I saw that issue, but I'm not using pip and my knowledge about Python is pretty much zero.. I installed pip, ran the command and started overviewer. It's running a top speed, so I believe is solved now.
But I thought that issue would be fixed in the latest version, but still applicable... Maybe mark as known error for new users?

I will close this issue, sorry for bothering! I should have tried the above command.
